docs: docs for machine id metrics and hw-health metrics fetch (#3377)
NVIDIA/bare-metal-manager-core: docs/operations/monitoring-health.md

healthy, NICo treats the sensor as healthy. In that case the BMC health state is
the authority.

### Hardware Entity Metrics

Beyond sensors, BMCs expose scalar values on Redfish `*Metrics` resources —
error counters, throttle durations, bandwidth utilization, power figures — that
have no sensor backing. The entity metrics collector polls these and exports
them as Prometheus series.

The collector is **disabled by default**. Adding the `[collectors.metrics]`

> section to the hardware health service config enables it:

Reviewer: It should be really obvious what collector we're talking about, but this is important, and good to have it stated like this for search results etc.

```suggestion
The entity metrics collector is **disabled by default**. Add the `[collectors.metrics]`
section to the hardware health service config to enable it:
```

```toml
[collectors.metrics]
fetch_interval = "2m"     # default
fetch_concurrency = 4     # default; parallel per-entity fetches
```

What it collects, per entity type discovered on the BMC:

| Redfish source | Examples |
|---|---|
| `ProcessorMetrics` | Core/other error counters, PCIe error counters (fatal, non-fatal, correctable, replay, NAK, bad TLP/DLLP), power/thermal throttle durations, bandwidth, frequency, temperature, consumed power, core voltage. |
| `MemoryMetrics` | Corrected volatile/persistent errors, current-period and lifetime ECC counters, dirty shutdowns, bandwidth, operating speed, capacity utilization. |
| `DriveMetrics` | Correctable/uncorrectable read and write I/O errors, bad blocks, power-on hours, read/write volume. |
| `PowerSupplyMetrics` | Input voltage/current/power, output power, energy, frequency, temperature, fan speed. |

> Values that are sensor-backed (carrying a Redfish `DataSourceUri`) are skipped:

Reviewer: ```suggestion
Sensor-backed values (carrying a Redfish `DataSourceUri`) are skipped:
```

the sensor collector already publishes them as `hw_sensor` series, so nothing
is double-reported.

Exported series are named
`{prefix}_hw_metric_{metric_type}_{unit}` — with the default
`carbide_hardware_health` prefix, for example:

```text
carbide_hardware_health_hw_metric_correctable_core_errors_count
carbide_hardware_health_hw_metric_pcie_fatal_errors_count
carbide_hardware_health_hw_metric_bandwidth_percent
carbide_hardware_health_hw_metric_input_power_watts
```

On the Prometheus endpoint, series carry entity labels (`processor_id`,
`memory_id`, `drive_id`, `powersupply_id`, `system_id`, `model`, ...) plus the
standard identity labels added by the sink (`machine_id`, `endpoint_ip`,
`serial_number`, `rack_id`, ...), with `collector_type="metrics_collector"`.

> The OTLP sink (`[sinks.otlp]`) emits the same metric *names*, but places the

Reviewer: Technically this can all be one paragraph, but easier to scan as two.

```suggestion

The OTLP sink (`[sinks.otlp]`) emits the same metric *names*, but places the
```

)
```

### Per-Object Health Metrics

The aggregate metrics above report *counts* of unhealthy objects. To identify
*which* objects carry a given health-alert classification, NICo can emit one
additional time series per affected object:

```text
carbide_object_unhealthy_by_classification_count{object_type="machine",object_id="fm100...",classification="Hardware",in_use="true"} 1
```

Labels:

| Label | Values |
|---|---|
| `object_type` | `machine`, `switch`, `rack`, `power_shelf` |
| `object_id` | The object's NICo id. |
| `classification` | The health-alert classification. |
| `in_use` | Machines only: whether a tenant instance uses the host. |

> Emission is opt-in per classification to contain cardinality: series count

Reviewer: Maybe "reduce data volume" instead of "contain cardinality," for a bit simpler language?

Author: In this exact case cardinality should be clearer. And it is not analogues to volume, you can have thousands of data points, but with low cardinality. And it is cardinality what is the problem here.

Also cardinality is the common term for the metrics. I feel what `reduce data volume` can be a little too cryptic when in fact we address cardinality.

still scales with fleet size (one series per matching object per listed
classification — an object carrying two enabled classifications emits two
series), but only for the classifications you list. It is disabled by
default; enable it in the NICo API config by listing the classifications to
emit:

```toml
[observability]
per_object_metrics_for_classifications = ["Hardware", "PreventAllocations"]
```

With an empty list (the default) the metric is not registered at all; the
aggregate health metrics are unaffected either way. Series disappear
automatically when the object becomes healthy, loses the classification, or
is deleted — entries are retained for the registry's hold period, which is
configured slightly longer than the state controllers' `metric_hold_time`.

Example: list hosts blocked from allocations by a hardware problem, or alert

> when hardware-unhealthy machines accumulate fleet-wide:

Reviewer: ```suggestion
For example, use the following PromQL query to list hosts blocked from allocations by a hardware problem, or alert when hardware-unhealthy machines accumulate fleet-wide:
```
